homeassistant/core.py

```diff
@@ -78,6 +78,18 @@ def valid_entity_id(entity_id: str) -> bool:
     return ENTITY_ID_PATTERN.match(entity_id) is not None
 
 
+def async_safe(func):
+    """Annotation to mark method as safe to call from within the event loop."""
+    # pylint: disable=protected-access
+    func._hass_async_safe = True
+    return func
+
+
+def is_async_safe(func):
```

Comment:
This would then be ``is_callback`` which seems less confusion-prone since coroutines are also 'async safe'.